### PR TITLE
Added documentation (GeometryInstance and VisualInstance)

### PR DESCRIPTION
**Pull request by TwistedTwigleg**
I've filled in the documentation for both GeometryInstance and VisualInstance.

### CODE REVIEW

File: doc/classes/GeometryInstance.xml [hunk 1]
```diff
@@ -14,32 +14,54 @@
 	</methods>
 	<members>
 		<member name="cast_shadow" type="int" setter="set_cast_shadows_setting" getter="get_cast_shadows_setting" enum="GeometryInstance.ShadowCastingSetting">
+			The currently selected shadow casting flag.
```
Comment (cbscribe) [inline doc/classes/GeometryInstance.xml]: Whenever possible, remove extra words that don't add content, ex: "currently".

When a member uses named constants, the standard has been to refer to them in the description, so the reader knows where to look. "See SHADOW_CASTING_SETTING_* constants for values."

File: doc/classes/GeometryInstance.xml [hunk 2]
```diff
@@ -14,32 +14,54 @@
...
 		</member>
 		<member name="extra_cull_margin" type="float" setter="set_extra_cull_margin" getter="get_extra_cull_margin">
+			The extra distance added to the GeometryInstance's bounding box ([AABB]) to increase it's cull box.
```
Comment (cbscribe) [inline doc/classes/GeometryInstance.xml]: it's -> its

File: doc/classes/GeometryInstance.xml [hunk 3]
```diff
@@ -14,32 +14,54 @@
...
 		</member>
 		<member name="lod_max_distance" type="float" setter="set_lod_max_distance" getter="get_lod_max_distance">
+			The max distance for the LOD of this GeometryInstance.
```
Comment (cbscribe) [inline doc/classes/GeometryInstance.xml]: Per doc guidelines, prefer possessive over "of the" constructions. 

The GeometryInstance's max LOD distance.

File: doc/classes/GeometryInstance.xml [hunk 4]
```diff
@@ -14,32 +14,54 @@
...
 		</member>
 		<member name="lod_min_distance" type="float" setter="set_lod_min_distance" getter="get_lod_min_distance">
+			The min distance for the LOD of this GeometryInstance.
 		</member>
 		<member name="lod_min_hysteresis" type="float" setter="set_lod_min_hysteresis" getter="get_lod_min_hysteresis">
+			The min margin for the LOD of this GeometryInstance.
 		</member>
 		<member name="material_override" type="Material" setter="set_material_override" getter="get_material_override">
+			The material override for the whole geometry.
+
```
Comment (akien-mga) [inline doc/classes/GeometryInstance.xml]: Please don't add empty lines, one line == one paragraph already. Use `git commit --amend` to modify that commit (see [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html)).

File: doc/classes/GeometryInstance.xml [hunk 5]
```diff
@@ -14,32 +14,54 @@
...
+			The currently selected shadow casting flag.
 		</member>
 		<member name="extra_cull_margin" type="float" setter="set_extra_cull_margin" getter="get_extra_cull_margin">
+			The extra distance added to the GeometryInstance's bounding box ([AABB]) to increase it's cull box.
 		</member>
 		<member name="lod_max_distance" type="float" setter="set_lod_max_distance" getter="get_lod_max_distance">
+			The max distance for the LOD of this GeometryInstance.
 		</member>
 		<member name="lod_max_hysteresis" type="float" setter="set_lod_max_hysteresis" getter="get_lod_max_hysteresis">
+			The max margin for the LOD of this GeometryInstance.
```
Comment (cbscribe) [inline doc/classes/GeometryInstance.xml]: See previous comment.

File: doc/classes/GeometryInstance.xml [hunk 6]
```diff
@@ -14,32 +14,54 @@
...
 		</member>
 		<member name="lod_min_distance" type="float" setter="set_lod_min_distance" getter="get_lod_min_distance">
+			The min distance for the LOD of this GeometryInstance.
```
Comment (cbscribe) [inline doc/classes/GeometryInstance.xml]: See previous comment.

File: doc/classes/GeometryInstance.xml [hunk 7]
```diff
@@ -14,32 +14,54 @@
...
 		</member>
 		<member name="lod_min_hysteresis" type="float" setter="set_lod_min_hysteresis" getter="get_lod_min_hysteresis">
+			The min margin for the LOD of this GeometryInstance.
```
Comment (cbscribe) [inline doc/classes/GeometryInstance.xml]: See previous comment.

File: doc/classes/GeometryInstance.xml [hunk 8]
```diff
@@ -14,32 +14,54 @@
...
 		</member>
 		<member name="material_override" type="Material" setter="set_material_override" getter="get_material_override">
+			The material override for the whole geometry.
+
+			If there is a material in material_override, it will be used instead of any material set in any material slot of the mesh.
```
Comment (cbscribe) [inline doc/classes/GeometryInstance.xml]: Don't use blank lines. Newlines start new paragraphs.

File: doc/classes/GeometryInstance.xml [hunk 9]
```diff
@@ -14,32 +14,54 @@
...
 		</member>
 		<member name="use_in_baked_light" type="bool" setter="set_flag" getter="get_flag">
+			If [code]true[/code], then this GeometryInstance will be used when baking lights using a [GIProbe] and/or any other form of baked lighting.
```
Comment (cbscribe) [inline doc/classes/GeometryInstance.xml]: Remove comma after "If [code]true[/code]"
Remove "then"

File: doc/classes/VisualInstance.xml [hunk 10]
```diff
@@ -13,12 +13,14 @@
...
 			</return>
 			<description>
+				Returns the [AABB] (also known as the bounding box) for this VisualInstance.
 			</description>
 		</method>
 		<method name="get_transformed_aabb" qualifiers="const">
 			<return type="AABB">
 			</return>
 			<description>
+				Returns the transformed [AABB] (also known as the bounding box) for this VisualInstance.
```
Comment (cbscribe) [inline doc/classes/VisualInstance.xml]: Might be useful to indicate what "transformed" means here.

File: doc/classes/VisualInstance.xml [hunk 11]
```diff
@@ -27,11 +29,17 @@
 			<argument index="0" name="base" type="RID">
 			</argument>
 			<description>
+				Allows you to change the base of te VisualInstance, which changes how the engine handels the VisualInstance under the hood.
```
Comment (cbscribe) [inline doc/classes/VisualInstance.xml]: te -> the
handels -> handles

"Allows you to change" -> Sets

File: doc/classes/VisualInstance.xml [hunk 12]
```diff
@@ -27,11 +29,17 @@
...
+
+				It is recommended to only use set_base if you know what you're doing.
 			</description>
 		</method>
 	</methods>
 	<members>
 		<member name="layers" type="int" setter="set_layer_mask" getter="get_layer_mask">
+			The render layer(s) this VisualInstance is drawn on.
+
+			This object will only be visible for [Camera]'s whose cull mask includes the render object this VisualInstance is set to.
```
Comment (cbscribe) [inline doc/classes/VisualInstance.xml]: "[Camera]'s" - remove apostrophe, this isn't possessive.

**Review by cbscribe — CHANGES_REQUESTED** [review]
Great work! I added some corrections and a couple of requests for clarification.

**Comment by TwistedTwigleg** [comment]
Thanks @cbscribe  and @akien-mga!

I fixed the problems in VisualInstance and GeometeryInstance. Now they hopefully have no more spelling and grammar issues, along with being more clear. I fixed them in the second commit, is that okay? I can try to fix the first commit if needed.

I added documentation for World, Theme, StreamTexture, Sky, and ShortCut. They probably could use a second pair of eyes, since I'm not sure everything is 100% accurate to what they actually do (mainly Sky and Shortcut). I tried to assure they don't have problems like VisualInstance and GeometeryInstance, so hopefully everything is good there.

**Comment by cbscribe** [comment]
You will need to squash down to one commit before this can be merged - the Godot team likes to keep the commit log clean.

Also, as a general rule of thumb, try and do a separate PR for different, unrelated changes. It's easier to track and review if they're not all in one gigantic commit.

**Comment by TwistedTwigleg** [comment]
Should I squash this commit into one? Or make another PR with the additional documentation? (World, Theme, StreamTexture, Sky, and ShortCut)

**Review by cbscribe — APPROVED** [review]

**Comment by cbscribe** [comment]
Thanks for your effort!

**Pull request merged**